tests/unit/test_session.py

```diff
@@ -119,13 +120,41 @@ def test_resolve_import_path_ignore_import_path(tmp_path_factory):
         os.remove(a_temp_file)
 
 
+@pytest.mark.parametrize("has_current_database", (True, False))
+def test_resolve_package_current_database(has_current_database):
+    def mock_get_current_parameter(param: str, quoted: bool = True) -> Optional[str]:
+        return "db" if has_current_database else None
+
+    def mock_get_information_schema_packages(table_name: str):
+        if (has_current_database and table_name == "information_schema.packages") or (
+            not has_current_database
+            and table_name == "snowflake.information_schema.packages"
+        ):
+            result = MagicMock()
+            result.filter().group_by().agg()._internal_collect_with_tag.return_value = [
+                ("random_package_name", json.dumps(["1.0.0"]))
+            ]
+            return result
```

Comment:
I understand that if the condition is False the test should fail with some exception because `mock_get_information_schema_packages` is None, but just to make it clear could we throw an exception in the else case or do some assertions at the top?

Reply:
Good suggestions :)